flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -234,21 +248,28 @@ public static String randomAlphaNumeric(int count) {
 
   public static class Result<T> {
     private final Future<T> future;
+    private final Consumer<Integer> exiter;
 
-    Result(Future<T> future) {
+    Result(Future<T> future, Consumer<Integer> exiter) {
```

Comment:
If we do it this way, we might just pass in an optional of hook and run it if it is not empty, followed by System.exit, so we don’t need to write that code twice and no need for the comment either. Or even better just pass in an iterator of optionals (of hooks) and loop through it, so we can keep the chain.

Reply:
Also I guess we want to use TerminationHook as argument type?